packages/ai-jsx/src/batteries/logging-integrations.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * Integrations with external logging services.
+ * Currently only Weights and Biases is supported.
+ * You can see examples package for an example with OpenTelemetry.
+ * @packageDocumentation
+ */
+import { AsyncLocalStorage } from 'node:async_hooks';
+import { v4 as uuidv4 } from 'uuid';
+import { StatusCode, WBTraceTree, addChildSpan, wandb } from '@nick.heiner/wandb-fork';
+import * as AI from '../index.js';
+import { debug } from '../core/debug.js';
+
+type WBSpan = Parameters<typeof addChildSpan>[0];
+
+function bindAsyncGenerator<T = unknown, TReturn = any, TNext = unknown>(
+  generator: AsyncGenerator<T, TReturn, TNext>
+): AsyncGenerator<T, TReturn, TNext> {
+  const result = {
+    next: AsyncLocalStorage.bind(generator.next.bind(generator)),
+    return: AsyncLocalStorage.bind(generator.return.bind(generator)),
+    throw: AsyncLocalStorage.bind(generator.throw.bind(generator)),
+
+    [Symbol.asyncIterator]() {
+      return result;
+    },
+  };
+
+  return result;
+}
+
+/**
+ * Exporting a forked version of wandb.
+ * You need to import this version of wandb to use the integration.
+ */
+export { wandb };
```

Comment:
@NickHeiner is there more info we should add here? E.g. "this is a temporary solution until X".
Also not sure if this is the right approach. I'm exporting this because when I tried importing it twice, it raised an error, so I did this.

Reply:
Ah, I was thinking this component would call `init` and `finish` for the user. @petersalas what do you think?

That way, we wouldn't have to export this, which is awkward.

Reply:
I'll take another look. The `init` I feel like I can do. For `finish` I'm not quite sure.

Reply:
Also what is the difference between your fork and the official W&B package? Are you going to create a PR for them?

Reply:
I pushed `init` inside, but I don't quite know how to do it for `finish` (since I don't 100% understand what is even going on here).

Reply:
From skimming their SDK it looks like it's effectively global, so I don't think we want to call `finish` within the render itself.

Can we have the consumer pass in the `wandb` object as a prop and let them take responsibility for `init`/`finish`?

Reply:
> Can we have the consumer pass in the `wandb` object

Hmm, is that so much better? It's not like `wandb.init()` creates a new object, right?

Also there's a question for Nick: What's the difference between your fork and the original? Potentially we can't let users supply their own `wandb` if it doesn't have have Nick needed to get it to work here.

Reply:
I think the things Nick had to change were largely exporting types...but since we end up just calling `toJSON()` on it I think we could construct the objects entirely on our side without even importing the SDK.

I think that's probably the best option here? It lets us avoid needing to rely on the fork and defers the lifetime questions to the consumer.

Reply:
Not sure I understand. Which object should we create then?
I'll setup a quick meeting to go over this. The conversation is dragging on and it'll probably take more of your time anyway.